openid/signingkeyprovider.go

```diff
@@ -70,8 +76,7 @@ func (s *signingKeyProvider) getSigningKey(issuer string, kid string) (interface
 }
 
 func findKey(km map[string][]signingKey, issuer string, kid string) []byte {
-	lock.RLock()
```

Comment:
@lonelycode I am merging pieces of this PR to the master branch...Do you recall what was wrong with having the lock.RLock() inside of the findKey func? I see you changed the approach but I dont see the issue with your previous one.